### ISSUE TITLE
CUDA_ERROR_ILLEGAL_ADDRESS: an illegal memory access was encountered

### ISSUE THREAD

**Issue by iou2much** [athena-team/athena#79]
Hi, I use mtl_transformer_sp.json for Fine-tuning stage. It finished the epoch 0, but error occurs at some point of epoch 1.
BTW, when I run it without `"speed_permutation": [0.9, 1.0, 1.1]`, it works. 

Here's the command:
`    horovodrun -np 4 -H localhost:4 python athena/horovod_main.py examples/asr/seewo/configs/mtl_transformer_sp.json
`
Here's the config:
`
{
  "batch_size":24,
  "num_epochs":50,
  "sorta_epoch":1,
  "ckpt":"examples/asr/mine/ckpts/mtl_transformer_ctc_sp/",
  "summary_dir":"examples/asr/mine/ckpts/mtl_transformer_ctc_sp/event",

  "solver_gpu":[0],
  "solver_config":{
    "clip_norm":100,
    "log_interval":10,
    "enable_tf_function":true
  },

  "model":"mtl_transformer_ctc",
  "num_classes": null,
  "pretrained_model": "examples/asr/mine/configs/mpc.json",
  "model_config":{
    "model":"speech_transformer",
    "model_config":{
      "return_encoder_output":true,
      "num_filters":512,
      "d_model":512,
      "num_heads":8,
      "num_encoder_layers":12,
      "num_decoder_layers":6,
      "dff":1280,
      "rate":0.1,
      "label_smoothing_rate":0.0,
      "schedual_sampling_rate":0.9
    },
    "mtl_weight":0.5
  },

  "decode_config":{
    "beam_search":true,
    "beam_size":10,
    "ctc_weight":0.5,
    "lm_weight":0.7,
    "lm_type": "rnn",
    "lm_path":"examples/asr/mine/configs/rnnlm.json"
  },

  "optimizer":"warmup_adam",
  "optimizer_config":{
    "d_model":512,
    "warmup_steps":25000,
    "k":1.0
  },

  "dataset_builder": "speech_recognition_dataset",
  "num_data_threads": 1,
  "trainset_config":{
    "data_csv": "examples/asr/mine/data/train.csv",
    "audio_config":{"type":"Fbank", "filterbank_channel_count":40},
    "cmvn_file":"examples/asr/mine/data/cmvn",
    "text_config": {"type":"vocab", "model":"examples/asr/mine/data/vocab"},
    "speed_permutation": [0.9, 1.0, 1.1],
    "input_length_range":[10, 8000]
  },
  "devset_config":{
    "data_csv": "examples/asr/mine/data/dev.csv",
    "audio_config":{"type":"Fbank", "filterbank_channel_count":40},
    "cmvn_file":"examples/asr/mine/data/cmvn",
    "text_config": {"type":"vocab", "model":"examples/asr/mine/data/vocab"},
    "input_length_range":[10, 8000]
  },
  "testset_config":{
    "data_csv": "examples/asr/mine/data/dev.csv",
    "audio_config":{"type":"Fbank", "filterbank_channel_count":40},
    "cmvn_file":"examples/asr/mine/data/cmvn",
    "text_config": {"type":"vocab", "model":"examples/asr/mine/data/vocab"}
  }
}

`
And here's the log:

> 
> [1,0]<stderr>:INFO:absl:global_steps: 15314	learning_rate: 1.7122e-04	loss: 3.6860	Accuracy: 0.8822	CTCAccuracy: 0.7879	sec/iter: 0.7007
> [1,0]<stderr>:INFO:absl:global_steps: 15324	learning_rate: 1.7133e-04	loss: 9.4535	Accuracy: 0.8708	CTCAccuracy: 0.7742	sec/iter: 0.6727
> [1,0]<stderr>:INFO:absl:global_steps: 15334	learning_rate: 1.7144e-04	loss: 5.3108	Accuracy: 0.8724	CTCAccuracy: 0.7839	sec/iter: 0.6807
> [1,0]<stderr>:INFO:absl:global_steps: 15344	learning_rate: 1.7155e-04	loss: 21.0516	Accuracy: 0.8447	CTCAccuracy: 0.7498	sec/iter: 0.6419
> [1,0]<stderr>:INFO:absl:global_steps: 15354	learning_rate: 1.7166e-04	loss: 11.5386	Accuracy: 0.8252	CTCAccuracy: 0.7421	sec/iter: 0.6913
> [1,0]<stderr>:INFO:absl:global_steps: 15364	learning_rate: 1.7177e-04	loss: 9.2761	Accuracy: 0.8529	CTCAccuracy: 0.7660	sec/iter: 0.7714
> [1,0]<stderr>:INFO:absl:global_steps: 15374	learning_rate: 1.7189e-04	loss: 6.6673	Accuracy: 0.8661	CTCAccuracy: 0.7800	sec/iter: 0.8484
> [1,0]<stderr>:INFO:absl:global_steps: 15384	learning_rate: 1.7200e-04	loss: 8.5238	Accuracy: 0.8668	CTCAccuracy: 0.7932	sec/iter: 0.8024
> [1,0]<stderr>:INFO:absl:global_steps: 15394	learning_rate: 1.7211e-04	loss: 10.3194	Accuracy: 0.8655	CTCAccuracy: 0.7854	sec/iter: 0.6500
> [1,0]<stderr>:INFO:absl:global_steps: 15404	learning_rate: 1.7222e-04	loss: 7.6797	Accuracy: 0.8471	CTCAccuracy: 0.7623	sec/iter: 0.6240
> [1,1]<stderr>:2020-03-28 02:04:52.312873: E tensorflow/stream_executor/cuda/cuda_event.cc:29] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS: an illegal memory access was encountered
> [1,1]<stderr>:2020-03-28 02:04:52.312922: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:273] Unexpected Event status: 1
> [1,1]<stderr>:[ef53b0d505e4:40836] *** Process received signal ***
> [1,1]<stderr>:[ef53b0d505e4:40836] Signal: Aborted (6)
> [1,1]<stderr>:[ef53b0d505e4:40836] Signal code:  (-6)
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 0] [1,1]<stderr>:/lib/x86_64-linux-gnu/libc.so.6(+0x3ef20)[0x7f223a69ef20]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 1] [1,1]<stderr>:/lib/x86_64-linux-gnu/libc.so.6(gsignal+0xc7)[0x7f223a69ee97]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 2] [1,1]<stderr>:/lib/x86_64-linux-gnu/libc.so.6(abort+0x141)[0x7f223a6a0801]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 3] [1,1]<stderr>:/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(+0x88d59b4)[0x7f2187b219b4]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 4] [1,1]<stderr>:/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZN10tensorflow8EventMgr10PollEventsEbPN4absl13InlinedVectorINS0_5InUseELm4ESaIS3_EEE+0x207)[0x7f2187a8d357]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 5] [1,1]<stderr>:/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZN10tensorflow8EventMgr8PollLoopEv+0x9f)[0x7f2187a8dbef]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 6] [1,1]<stderr>:/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/../libtensorflow_framework.so.2(_ZN5Eigen15ThreadPoolTemplIN10tensorflow6thread16EigenEnvironmentEE10WorkerLoopEi+0x281)[0x7f217e5718b1]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 7] [1,1]<stderr>:/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/../libtensorflow_framework.so.2(_ZNSt17_Function_handlerIFvvEZN10tensorflow6thread16EigenEnvironment12CreateThreadESt8functionIS0_EEUlvE_E9_M_invokeERKSt9_Any_data+0x48)[0x7f217e56efa8]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 8] [1,1]<stderr>:/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/../libtensorflow_framework.so.2(+0x167b7cf)[0x7f217ebc87cf]
> [1,1]<stderr>:[ef53b0d505e4:40836] [ 9] /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db)[0x7f223a4486db]
> [1,1]<stderr>:[ef53b0d505e4:40836] [10] [1,1]<stderr>:/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f)[0x7f223a78188f]
> [1,1]<stderr>:[ef53b0d505e4:40836] *** End of error message ***
> --------------------------------------------------------------------------
> Primary job  terminated normally, but 1 process returned
> a non-zero exit code. Per user-direction, the job has been aborted.
> --------------------------------------------------------------------------
> --------------------------------------------------------------------------
> mpirun noticed that process rank 1 with PID 0 on node ef53b0d505e4 exited on signal 6 (Aborted).
> --------------------------------------------------------------------------

**Comment by Some-random** [comment]
It looks more like a system issue with GPU card/ CUDA kernal. Please tell us your CUDA/cuDNN version and tensorflow version.

Also, please remove everything under folder ~/.nv/ and run mtl_transformer_sp.json again. Let me know if this problem still exists.

**Comment by iou2much** [comment]
Thanks a lot for the quick reply.
Here's more info.
```
root@ef53b0d505e4:/code/athena# pip list|grep tensor
tensorboard          2.0.2
tensorflow-estimator 2.0.1
tensorflow-gpu       2.0.1
```

The NVidia driver version is 440.31

```
root@ef53b0d505e4:/code/athena# nvcc  --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2018 NVIDIA Corporation
Built on Sat_Aug_25_21:08:01_CDT_2018
Cuda compilation tools, release 10.0, V10.0.130
```

```
root@ef53b0d505e4:/code/athena# apt list|grep -i cudnn
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
libcudnn7/unknown 7.6.5.32-1+cuda10.2 amd64 [upgradable from: 7.6.2.24-1+cuda10.0]
libcudnn7-dev/unknown 7.6.5.32-1+cuda10.2 amd64

```

**Comment by iou2much** [comment]
My ~/.nv is empty, but I clear it anyway.
And seems I may run into this error randomly. I can successfully train the 4th epoch now, and the error occur during the training, I just restart the run.sh and it can go on
